extensions/data-workspace/src/common/utils.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fs from 'fs';
+
+export async function directoryExist(directoryPath: string): Promise<boolean> {
+	const stats = await getFileStatus(directoryPath);
+	return stats ? stats.isDirectory() : false;
+}
+
+export async function fileExist(filePath: string): Promise<boolean> {
+	const stats = await getFileStatus(filePath);
```

Comment:
a lot of extensions have an exists function like this: https://github.com/microsoft/azuredatastudio/blob/ef0a92d83ffe8b169f50aba1e9ef7f77eb94491a/extensions/agent/src/agentUtils.ts#L52-L53

Is there a difference between using that and the one here?

Reply:
yes, we don't have a good way to share code between extensions. for this specific method, my requirement is more strict, I need to make sure it is a directory or file.